src/gui/src/viewer/zoom-window.cpp

```diff
@@ -74,6 +75,44 @@ ZoomWindow::ZoomWindow(QList<QSharedPointer<Image>> images, const QSharedPointer
 	QShortcut *toggleFullscreen = new QShortcut(QKeySequence::FullScreen, this);
 		connect(toggleFullscreen, &QShortcut::activated, this, &ZoomWindow::toggleFullScreen);
 
+	
+	switch (m_settings->value("actionButtons").toInt()) {	// Consider adding this as class variable.
+		case ButtonVisibility::All :	// Handled later by resizeButtons.
+			break;
+		case ButtonVisibility::None :
+			ui->buttonPlus->setVisible(false);
+			ui->actionButtons->setVisible(false);
+			break;
+		case ButtonVisibility::Favorites :
+			ui->drawer_1->setVisible(true);
+			ui->buttonPlus->setVisible(false);
+			ui->drawer_0->setVisible(false);
+
+			ui->drawer_0->setFixedSize(0, 0);	// Occupy zero width so navigation buttons can converge.
+
+			// Margin may have been set dynamically.
+			//ui->drawer_1->setContentsMargins(0, 0, 0, 0);	// This method does not seem to work:
+			//	log(std::to_string(ui->drawer_1->contentsMargins().left()).c_str());
```

Comment:
FYI, if you want to log something else than a string using `log()`, you can simply wrap it with `QString()`. For numbers, you have the [QString::number()](https://doc.qt.io/qt-5/qstring.html#number-1) static method to cast them.